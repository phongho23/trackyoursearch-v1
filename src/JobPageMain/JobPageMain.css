.JobPageMain {
  background-color: lightpink;
  height: 100%;
  padding: 10px;
  width: 100%;
  margin: auto;
  align-content: center;
  align-items: center;
}

.JobPageMain__jobTitle {
  margin: auto;
  align-content: center;
  justify-content: center;
}

.JobPageMain__companyName {
  margin: auto;
  align-content: center;
  justify-content: center;
}

.JobPageMain__interview {
  margin: auto;
  align-content: center;
  justify-content: center;
}

.JobPageMain__postedUrl {
  margin: auto;
  align-content: center;
  justify-content: center;
}

.JobPageMain__jobRating {
  margin: auto;
  align-content: center;
  justify-content: center;
}

.JobPageMain__content {
  margin: auto;
  align-content: center;
  justify-content: center;
}



.jobInfo {
  margin: auto;
  align-content: center;
  align-items: center;
}

.JobPageMain__content {
  color: #3a3939;
  max-width: 380px;
}

.JobPageMain__jobTitle {
  color: #3a3939;
  max-width: 380px;
  padding-top: 12px;
}

.JobPageMain__companyName {
  color: #3a3939;
  max-width: 380px;
  padding-top: 12px;
}

.JobPageMain__interview {
  color: #3a3939;
  max-width: 380px;
  padding-top: 12px;
}

.JobPageMain__jobRating {
  color: #3a3939;
  max-width: 380px;
  padding-top: 12px;
}

.JobPageMain__postedUrl {
  color: #3a3939;
  max-width: 380px;
  padding-top: 12px;
}
