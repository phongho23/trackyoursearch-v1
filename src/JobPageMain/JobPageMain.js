

import React from 'react'
import Jobitem from '../Jobitem/Jobitem'
import './JobPageMain.css'

export default function JobPageMain(props) {
  return (
    <section className='JobPageMain'>
      <Jobitem
        id={props.job.id}
        name={props.job.name}
        modified={props.job.modified}
      />
      <div className='jobInfo'>
      <div className='JobPageMain__jobTitle'>
        <b><p>Job Title:</p></b>
        {props.job.jobTitle}
      </div>

      <div className='JobPageMain__companyName'>
        <b><p>Company Name:</p></b>
        {props.job.companyName}
      </div>

      <div className='JobPageMain__postedUrl'>
        <b><p>Job Listing URL/Info:</p></b>
        {props.job.postedUrl}
      </div>

      <div className='JobPageMain__interview'>
        <b><p>Interview Details:</p></b>
        {props.job.interview}
      </div>

      <div className='JobPageMain__jobRating'>
        <b><p>Opportunity Rating:</p></b>
        {props.job.jobRating}
      </div>

      <div className='JobPageMain__content'>
        <b><p>Job Description:</p></b>
        {props.job.content.split(/\n \r|\n/).map((para, i) =>
          <p key={i}>{para}</p>
        )}
      </div>
      </div>
    </section>
  )
}

JobPageMain.defaultProps = {
  job: {
    content: '',
  }
}