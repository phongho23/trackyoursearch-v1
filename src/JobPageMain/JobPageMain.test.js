import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import JobPageMain from './JobPageMain'

describe(`JobPageMain component`, () => {
  const props = {
    job: {
      "id": "cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1",
      "name": "Dogs",
      "modified": "2019-01-03T00:00:00.000Z",
      // "weekId": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
      "content": "Corporis accusamus placeat.\n \rUnde."
    }
  }

  it('renders a .JobPageMain by default', () => {
    const wrapper = shallow(<JobPageMain />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders a Jobitem with job prop', () => {
    const job = shallow(<JobPageMain {...props} />)
      .find('Jobitem')
    expect(toJson(job)).toMatchSnapshot()
  })

  it(`splits the content by \\n or \\n\\r, with a p foreach`, () => {
    [{
      job: { "content": "Content with n r.\n \rafter n r." }
    }, {
      job: { "content": "Content with n.\nafter." }
    }].forEach(props => {
      const content = shallow(<JobPageMain {...props} />)
        .find('JobPageMain__content')
      expect(toJson(content)).toMatchSnapshot()
    })
  })
})
