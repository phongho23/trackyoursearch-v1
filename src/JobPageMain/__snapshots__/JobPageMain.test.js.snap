// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobPageMain component renders a .JobPageMain by default 1`] = `
<section
  className="JobPageMain"
>
  <Jobitem />
  <div
    className="JobPageMain__content"
  >
    <p
      key="0"
    />
  </div>
</section>
`;

exports[`JobPageMain component renders a Jobitem with job prop 1`] = `
<Jobitem
  id="cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1"
  modified="2019-01-03T00:00:00.000Z"
  name="Dogs"
/>
`;

exports[`JobPageMain component splits the content by \\n or \\n\\r, with a p foreach 1`] = `null`;

exports[`JobPageMain component splits the content by \\n or \\n\\r, with a p foreach 2`] = `null`;
